persistence/src/main/java/de/cyface/persistence/MeasurementTable.java

```diff
@@ -140,14 +152,106 @@ public void onUpgrade(final SQLiteDatabase database, final int fromVersion, fina
 
             case 15:
                 // This column was added in version 16
-                Log.d(TAG, "Upgrading event table from V15");
+                Log.d(TAG, "Upgrading measurement table from V15");
                 migrateDatabaseFromV15(database);
 
                 break; // onUpgrade is called incrementally by DatabaseHelper
+
+            case 16:
+                // These columns were added in version 17
+                Log.d(TAG, "Upgrading measurement table from V16");
+                migrateDatabaseFromV16(database);
+
+                break;
         }
 
     }
 
+    /**
+     * Adds columns related to average speed to table and calculates the values for existing data.
+     * <p>
+     * Like for newly captured measurements, the average speed is only based on locations not filtered
+     * by the {@link DefaultLocationCleaningStrategy}.
```

Comment:
I think it's expected, that the average speed does not decrease when the user is not moving for a short time. (also, because the customer said "should be like distance calculation)
Thus, I'm using the location cleaning strategy here in the data migration and in the actual data capturing before calculating average speed, to filter "invalid" locations.